test/test_prims.py

```diff
@@ -113,6 +113,122 @@ def _wrapper(a):
             self.assertTrue(result.is_contiguous)
             self.assertEqual(_wrapper(a), result)
 
+    @onlyCUDA
+    @skipCUDAIfRocm
+    @dtypes(torch.float32)
+    def test_copy_to(self, device, dtype):
+        from torch.fx.experimental.proxy_tensor import make_fx
+        from torch._prims.context import TorchRefsNvfuserCapabilityMode
+        from torch._prims.executor import execute
+
+        def func1(a, b):
+            return (a.copy_(b),)
+
+        def func2(a, b):
+            b_sin = b.sin()
+            return (a.copy_(b_sin),)
+
+        def func3(a, b):
+            return (a.copy_(b), b)
+
+        def func4(a, b):
+            b_sin = b.sin()
+            return (a.copy_(b_sin), b_sin)
+
+        def func5(a, b):
+            b_sin = b.sin()
+            a1 = a.copy_(b_sin)
+            a1_sin = a1.sin()
+            a2 = a1.copy_(a1_sin)
+            return (a2, b_sin, a1_sin)
+
+        for func in (func1, func3, func2, func3, func4, func5):
+            a = torch.empty(3, 3, device=device, dtype=dtype)
+            b = torch.randn(3, 3, device=device, dtype=dtype)
+
+            with TorchRefsNvfuserCapabilityMode():
+                gm = make_fx(func)(a, b)
+
+            a = torch.empty(3, 3, device='cuda')
+            out = execute(gm, a, b, executor="strictly_nvfuser")
+            self.assertEqual(out[0], a)
+
+            a = torch.empty(3, 3, device='cuda')
+            self.assertEqual(out[0], func(a, b)[0])
```

Comment:
I also feel the test is too relaxed and sends out a misleading message on what codegen alias supports.

We should be explicit on that the alias support right now is VERY limited. In the test example here, we should check that fusion maintains consistent behavior. We should check matching result on all outputs, as well as identical aliases among outputs and inputs.

Reply:
> Didn't nvfuser give up on aliasing inputs and outputs? (We had this discussion in transpose PR)

Yeah, we sorta did.
Our assumption is that functionalization pass will resolve this for us and we do not need to "return" outputs with correct alias.
Since test here doesn't use functionalization pass, I'm trying to keep tests cleaner.